File: LegacySource/settlement/room/spirit/dump/ROOM_DUMP.cs
using System;
using System.IO;
using settlement.misc.util;
using settlement.path.finders;
using settlement.room.main;
using settlement.room.main.category;
using settlement.room.main.furnisher;
using settlement.room.main.util;
using settlement.room.service.module;
using settlement.thing.ThingsCorpses;
using snake2d.util.file;
using util.text;

namespace settlement.room.spirit.dump
{
    public class ROOM_DUMP : RoomBlueprintIns<DumpInstance>, ROOM_SERVICE_HASER
    {
        private Constructor constructor;
        private RoomService service;
        public static string ¤¤RemoveProblem = "¤This resting place still holds the dead and can not be removed. Deactivate the room and allow the corpses to decompose peacefully. Current cadavers: {0}. Days until clear: {1}.";

        static
        {
            D.ts(typeof(ROOM_DUMP));
        }

        public ROOM_DUMP(RoomInitData data, RoomCategorySub cat) : base(0, data, "_DUMP_CORPSE", cat)
        {
            constructor = new Constructor(this, data);
            service = new RoomService(this, data, null)
            {
                public FSERVICE service(int tx, int ty)
                {
                    return Dump.get(tx, ty);
                }
            };
        }

        protected override void saveP(FilePutter saveFile)
        {
            service.saver.save(saveFile);
        }

        protected override void loadP(FileGetter saveFile)
        {
            service.saver.load(saveFile);
        }

        protected override void clearP()
        {
            service.saver.clear();
        }

        protected override void update(double ds)
        {
            // TODO Auto-generated method stub
        }

        public SFinderRoomService service(int tx, int ty)
        {
            return service.finder;
        }

        public RoomService service()
        {
            return service;
        }

        public Furnisher constructor()
        {
            return constructor;
        }

        public void burry(Corpse corpse, int tx, int ty)
        {
            Dump.get(tx, ty).burry(corpse);
        }
    }
}

File: settlement/room/spirit/dump/ROOM_DUMP.java
package settlement.room.spirit.dump;

import java.io.IOException;

import settlement.misc.util.FSERVICE;
import settlement.path.finders.SFinderRoomService;
import settlement.room.main.RoomBlueprintIns;
import settlement.room.main.category.RoomCategorySub;
import settlement.room.main.furnisher.Furnisher;
import settlement.room.main.util.RoomInitData;
import settlement.room.service.module.RoomService;
import settlement.room.service.module.RoomService.ROOM_SERVICE_HASER;
import settlement.thing.ThingsCorpses.Corpse;
import snake2d.util.file.FileGetter;
import snake2d.util.file.FilePutter;
import util.text.D;

public class ROOM_DUMP extends RoomBlueprintIns<DumpInstance> implements ROOM_SERVICE_HASER{

	final Constructor constructor;
	final RoomService service;
	public static CharSequence ¤¤RemoveProblem = "¤This resting place still holds the dead and can not be removed. Deactivate the room and allow the corpses to decompose peacefully. Current cadavers: {0}. Days until clear: {1}.";
	
	static {
		D.ts(ROOM_DUMP.class);
	}
	
	public ROOM_DUMP(RoomInitData data, RoomCategorySub cat) throws IOException {
		super(0, data, "_DUMP_CORPSE", cat);
		constructor = new Constructor(this, data);
		service = new RoomService(this, data, null) {
			
			@Override
			public FSERVICE service(int tx, int ty) {
				return Dump.get(tx, ty);
			}
		};
	}

	@Override
	protected void saveP(FilePutter saveFile) {
		service.saver.save(saveFile);
		
	}

	@Override
	protected void loadP(FileGetter saveFile) throws IOException {
		service.saver.load(saveFile);
		
	}

	@Override
	protected void clearP() {
		service.saver.clear();
	}

	@Override
	protected void update(double ds) {
		// TODO Auto-generated method stub
		
	}

	@Override
	public SFinderRoomService service(int tx, int ty) {
		return service.finder;
	}
	
	@Override
	public RoomService service() {
		return service;
	}

	@Override
	public Furnisher constructor() {
		return constructor;
	}


	
	public void burry(Corpse corpse, int tx, int ty) {
		Dump.get(tx, ty).burry(corpse);
	}

	
	
}
